src/design-system/components/icon-button/IconButton.tsx

```diff
@@ -0,0 +1,63 @@
+import { CircularProgress, IconButtonProps } from '@mui/material'
+import React, { useState } from 'react'
+import { IconButtonVariant } from '~/types'
+import AddRoundedIcon from '@mui/icons-material/AddRounded'
+import { IconButton as MuiIconButton } from '@mui/material'
+import './IconButton.scss'
+interface S2SIconButtonProps extends Omit<IconButtonProps, 'size'> {
+  variant?: IconButtonVariant
+  size?: 'xs' | 'sm' | 'md' | 'lg'
+  loading?: boolean
+  disabled?: boolean
+  toggleAble?: boolean
+}
+
+export const IconButton: React.FC<S2SIconButtonProps> = ({
+  variant = IconButtonVariant.Primary,
```

Comment:
I am not sure, but I think you should not use enums which are outside of design-system folder, because this basic component will be moved to another repository later.

Reply:
To be more consistent with existing solutions I think it is better to use common.enums.ts. For better approach I can move  my enum to design-system folder.